test/lint/lint-git-commit-check.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+# Copyright (c) 2017-2019 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+# Linter to check that commit messages have a new line before the body
+# or no body at all
+#
+# reference: gist.github.com/agnivade/67b42d664ece2d4210c7
+#
+# WARNING: Using a "+" character in commit messages can result in 
+# improper parsing
+
+export LC_ALL=C
+while read -d "+" -r msg  || [[ -n "$msg" ]]; do
+    n_line=0
+    while IFS= read -r line || [[ -n "$line" ]]; do
+        n_line=$((n_line+1))
+        length=${#line}
+        if [ $n_line -eq 2 ] && [ $length -ne 0 ]; then
+            echo "Your subject line follows a non-empty line. Subject lines should always be followed by a blank line."
+            exit 1
+        fi
+    done < <(echo "$msg")
+done < <(git log origin..HEAD --format=%s%b+)
```

Comment:
`$ git log origin..HEAD --format=%s%b+` isn't working for me while testing locally. I get:
```
fatal: ambiguous argument 'origin..HEAD': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

Reply:
I changed it to `git log origin/master..HEAD`. My intent is to get the last N commits that have not yet been pushed in a portable way. I did some research and I believe the change that I have made achieves this, but please let me know if not.